packages/components/src/DateTimePickers/DateTimePicker/DateTimePicker.component.js

```diff
@@ -28,12 +35,35 @@ class DateTimePicker extends React.Component {
 		this.onSelectTime = this.onSelectTime.bind(this);
 	}
 
+	componentWillReceiveProps(nextProps) {
+		const isSelectedDateUpdated = this.props.selectedDate !== nextProps.selectedDate;
+		const isSelectedTimeUpdated = this.props.selectedTime !== nextProps.selectedTime;
+
+		if (isSelectedDateUpdated || isSelectedTimeUpdated) {
+			const canUpdateCalendar = isSelectedDateUpdated && nextProps.selectedDate !== undefined;
+			this.setState({
+				...(isSelectedDateUpdated && { selectedDate: nextProps.selectedDate }),
+				...(isSelectedTimeUpdated && { selectedTime: nextProps.selectedTime }),
+				...(canUpdateCalendar && {
```

Comment:
In fact shouldn't we consider that if a props date or time change, it means that the whole selection should be updated ?
In storybook testing, I bumped on weird things : 
- empty the input
- select the hour, it doesn't trigger the onChange, that's ok
- type a date, the selected date is updated in the picker
I end up with date and hour selected in the picker, but the input doesn't reflect that.

If you update selected date and time in state everytime that one of them change, you'll have the behavior where user time, it automatically invalidate previous selection. Everything stay up to date.
In summary, any date-picker submit updates all the input, and any input change updates all the selection

Reply:
I think we are aligned. 
But don't add any date/time as props. You must have a clean api, don't duplicate the Date props with string equivalent for change detection. Just consider that if the Date prop ref change, you have a new Date. Or if you really want to avoid refresh, implement `shouldComponentUpdate()` to compare old and new Date.

Reply:
✔